kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -58,14 +61,30 @@ class FlinkProcessBuilder(
   // flink.execution.target are required in Kyuubi conf currently
   val executionTarget: Option[String] = conf.getOption("flink.execution.target")
 
+  private lazy val proxyUserEnable: Boolean = {
+    doAsEnabled && conf.get(ENGINE_FLINK_DOAS_ENABLED) &&
+    conf.getOption(s"$FLINK_CONF_PREFIX.$FLINK_SECURITY_KEYTAB_KEY").isEmpty &&
```

Comment:
in Spark, when doAs is enabled, we actually have a constraint to ensure that the principal should always be the session user.

Reply:
The implementation of flink proxy user is different from the built-in `--proxy-user` of spark. It relies on `HADOOP_PROXY_USER` and has some limitations. It is difficult for us to control the behavior of the client well.

Reply:
we should unify the concept on the Kyuubi layer as much as possible.